worker/instancemutater/mutater.go

```diff
@@ -106,13 +109,16 @@ func (m *mutater) startMachines(tags []names.MachineTag) error {
 	return nil
 }
 
-func runMachine(machine MutaterMachine, removed <-chan struct{}, died chan<- instancemutater.MutaterMachine) {
+func runMachine(machine MutaterMachine, removed <-chan struct{}, died chan<- instancemutater.MutaterMachine, dying <-chan struct{}, done func()) {
+	defer done()
 	defer func() {
 		// We can't just send on the dead channel because the
 		// central loop might be trying to write to us on the
 		// removed channel.
 		for {
 			select {
+			case <-dying:
```

Comment:
suggest:
case <- machine.context.dying()

The machines and the worker share a context, no need to pass it in. It's buried down in mutaterWorker newMachineContext().  Ran the race stress 500+ successfully.